tests/Unit/Domain/Test_ElementDistribution.cpp

```diff
@@ -3,306 +3,780 @@
+      "  - LowerCornerIndex: [2, 1]\n"
+      "    UpperCornerIndex: [3, 3]\n"
+      "    Refinement: [6, 7]");
+
+  const auto domain = domain_creator->create_domain();
+  const auto& blocks = domain.blocks();
+
+  const auto costs = domain::get_element_costs(
+      blocks, domain_creator->initial_refinement_levels(),
+      domain_creator->initial_extents(), domain::ElementWeight::Uniform,
+      std::nullopt);
+
+  for (const auto& element_id_and_cost : costs) {
+    CHECK(element_id_and_cost.second == 1.0);
   }
-  return element_index;
 }
 
-template <size_t Dim>
-size_t number_of_elements_in_block(
-    const std::array<size_t, Dim>& refinement_levels) {
-  size_t number_of_elements = 1;
-  for (size_t i = 0; i < Dim; ++i) {
-    number_of_elements *= two_to_the(gsl::at(refinement_levels, i));
+// Test the weighting done by `domain::get_element_costs` for weighted cost
+// functions
+void test_weighted_cost_function(const domain::ElementWeight element_weight) {
+  const auto domain_creator1 = make_domain_creator<3>(
+      std::string{"AlignedLattice:\n"
+                  "  BlockBounds: [[0.0, 1.0, 2.0], [0.0, 1.0], [0.0, 1.0]]\n"
+                  "  IsPeriodicIn: [false, false, false]\n"
+                  "  InitialGridPoints: [4, 4, 4]\n"
+                  "  InitialLevels: [2, 1, 0]\n"
+                  "  RefinedLevels: []\n"
+                  "  RefinedGridPoints: []\n"
+                  "  BlocksToExclude: []\n"});
+  const auto domain1 = domain_creator1->create_domain();
+  const auto& blocks1 = domain1.blocks();
+
+  // Block size and grid points are the same as blocks in `domain1`, but
+  // refinement levels are different
+  const auto domain_creator2 = make_domain_creator<3>(
+      std::string{"AlignedLattice:\n"
+                  "  BlockBounds: [[0.0, 1.0], [0.0, 1.0], [0.0, 1.0]]\n"
+                  "  IsPeriodicIn: [false, false, false]\n"
+                  "  InitialGridPoints: [4, 4, 4]\n"
+                  "  InitialLevels: [2, 3, 2]\n"
+                  "  RefinedLevels: []\n"
+                  "  RefinedGridPoints: []\n"
+                  "  BlocksToExclude: []\n"});
+  const auto domain2 = domain_creator2->create_domain();
+  const auto& blocks2 = domain2.blocks();
+
+  // Block size and refinement levels are the same as blocks in `domain1`, but
+  // grid points are different
+  const auto domain_creator3 = make_domain_creator<3>(
+      std::string{"AlignedLattice:\n"
+                  "  BlockBounds: [[0.0, 1.0], [0.0, 1.0], [0.0, 1.0]]\n"
+                  "  IsPeriodicIn: [false, false, false]\n"
+                  "  InitialGridPoints: [4, 3, 2]\n"
+                  "  InitialLevels: [2, 1, 0]\n"
+                  "  RefinedLevels: []\n"
+                  "  RefinedGridPoints: []\n"
+                  "  BlocksToExclude: []\n"});
+  const auto domain3 = domain_creator3->create_domain();
+  const auto& blocks3 = domain2.blocks();
+
+  const auto costs1 = domain::get_element_costs(
```

Comment:
I ended up not doing this because I found that there wasn't really a nice and clean way to do it because the `if` / `else` needs access to the 3 costs computed, which would be local to the `for` loop, so if I put the cost computation in the loop, I would need to compute them again for the `if` / `else` after. But please let me know if there is something I'm not seeing/understanding!